datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/licenses.py

```diff
@@ -310,35 +306,42 @@ def parse_license_path(tar_file_name):
     return None
 
 
-def generate_tarfiles(tar_path):
-    """
-    Streams tarball archive and generates tarfile for each file.
-    """
-    stream = requests.get(tar_path, stream=True)
-    with tarfile.open(fileobj=stream.raw, mode="r|gz") as tar_file:
-        for tar_info in tar_file:
-            if not tar_info.islnk() and not tar_info.issym():
-                yield tar_info, tar_file
-    stream.close()
+def generate_from_tarball(stream):
+    with tarfile.open(fileobj=stream, mode="r|gz") as archive:
+        for tar_info in archive:
+            if not (tar_info.islnk() or tar_info.issym()) and parse_license_path(tar_info.name):
+                fh = archive.extractfile(tar_info)
+                if fh is not None:
+                    with closing(fh) as safe_fh:
+                        yield safe_fh.read()
+
 
+def generate_from_wheel(stream):
+    with ZipFile((stream)) as archive:
```

Comment:
I'm not sure why the linter is not catching this:

```suggestion
    with ZipFile(stream) as archive:
```

Reply:
Oh, I'm surprised our auto-formatting didn't catch this. Thanks!